### PR TITLE
Warn-level logging when an operator finds deprecated or unknown properties 

### PR DESCRIPTION
**Pull request by tombentley**
### Type of change

- Enhancement / new feature

### Description

Warn-level logging when an operator finds deprecated or unknown properties 

Also make the CrdGenerator error when a PoJo class does not implement
UnknownPropertyPreserving, or does not appear to have overridden hashCode(),
equals().

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Update/write design documentation in `./design`
- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: examples/kafka/kafka-ephemeral.yaml [hunk 1]
```diff
@@ -4,6 +4,7 @@ metadata:
   name: my-cluster
 spec:
   kafka:
+    foo: bar
```
Comment (scholzj) [inline examples/kafka/kafka-ephemeral.yaml]: I think this should not be here!

File: examples/kafka/kafka-ephemeral.yaml [hunk 2]
```diff
@@ -20,6 +21,4 @@ spec:
     replicas: 3
     storage:
       type: ephemeral
-  entityOperator:
-    topicOperator: {}
-    userOperator: {}
+  topicOperator: {}
```
Comment (scholzj) [inline examples/kafka/kafka-ephemeral.yaml]: I think this should not be here!

File: examples/kafka/kafka-ephemeral.yaml (hunk 1 above)
Comment (tombentley) [reply examples/kafka/kafka-ephemeral.yaml]: Oh yeah, good point, thanks ☺

File: examples/kafka/kafka-ephemeral.yaml (hunk 2 above)
Comment (scholzj) [reply examples/kafka/kafka-ephemeral.yaml]: @tombentley Your last commit removed the `topicOperator`. But is didn't revert the removal of the `entityOperator`. We have to add that back.

**Comment by scholzj** [comment]
@tombentley Is this worth adding to `CHANGELOG.md` and to the roadmap? I think this is fairly important improvement.

**Pull request merged**